### PR TITLE
[main] Initial `cgmanifest.json` clean-up.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Initial clean-up of the `cgmanifest.json` file. Mostly URL fixes but also additions of a few missing ones.

###### Change Log  

- Fixed/Added entries inside `cgmanifest.json`.
- Updated a few specs to point to a working URL.
- Updated the error message from the `updates_cgmanifest.py` script.

###### Does this affect the toolchain?  

No.

###### Test Methodology

- Local URL download checks.

### CODE REVIEW

File: .github/workflows/validate-cg-manifest.sh [hunk 1]
```diff
@@ -125,10 +125,11 @@ do
     else
       # Try a few times to download the source listed in the manifest
       mkdir -p ./cgmanifest_test_dir
-      for _ in {1..10}
+      for attempt in {1..10}
       do
         wget --quiet -P ./cgmanifest_test_dir "$manifesturl" && touch ./cgmanifest_test_dir/WORKED && break
```
Comment (oliviacrain) [inline .github/workflows/validate-cg-manifest.sh:130]: Doesn't need to be in this PR, but we may want to consider using the `--spider` option with `wget` to check for a URL's existence without downloading the file. Could save us a decent amount of time.

File: cgmanifest.json [hunk 2]
```diff
@@ -3389,7 +3399,7 @@
         "other": {
           "name": "expat",
           "version": "2.4.3",
-          "downloadUrl": "https://github.com/libexpat/libexpat/releases/download/R_2_4_3/expat-2.4.3.tar.bz2"
+          "downloadUrl": "https://github.com/libexpat/libexpat/releases/download/R_$(echo"
```
Comment (oliviacrain) [inline cgmanifest.json:3402]: This URL seems to be invalid

File: cgmanifest.json [hunk 3]
```diff
@@ -5251,7 +5271,7 @@
         "other": {
           "name": "harfbuzz",
           "version": "3.4.0",
-          "downloadUrl": "https://github.com/harfbuzz/harfbuzz/releases/download/3.4.0/harfbuzz-3.4.0.tar.xz"
+          "downloadUrl": "https://github.com/harfbuzz/harfbuzz/releases/download/3.3.2/harfbuzz-3.4.0.tar.xz"
```
Comment (oliviacrain) [inline cgmanifest.json:5274]: Getting a 404 for this URL

**Review by oliviacrain — CHANGES_REQUESTED** [review]
Looks good overall- left comments on a couple of URL changes that seem invalid.

File: cgmanifest.json (hunk 2 above)
Comment (PawelWMS) [reply cgmanifest.json:3402]: Good catch, thanks!

File: cgmanifest.json (hunk 3 above)
Comment (PawelWMS) [reply cgmanifest.json:5274]: Another one I missed, thanks, Olivia!

File: .github/workflows/validate-cg-manifest.sh (hunk 1 above)
Comment (PawelWMS) [reply .github/workflows/validate-cg-manifest.sh:130]: Thanks for the hint! I'm actually hoping to do some improvements to this script as part of the `cgmanifest.json` clean-up work and making the checks more efficient is one of them, so this should fit nicely into that.:)

**Review by oliviacrain — APPROVED** [review]

**Pull request merged**